packages/api/src/controllers/v1/agit/mod.rs

```diff
@@ -81,14 +101,23 @@ impl AgitControllerV1 {
         Json(body): Json<AgitByIdAction>,
     ) -> Result<Json<Agit>> {
         tracing::debug!("agit act_by_id {id} {body:?}");
+
         match body {
-            AgitByIdAction::Update(_) => {
-                //TODO: Add Update Agit
-                Ok(Json(Agit::default()))
+            AgitByIdAction::Update(req) => {
+                let update_request = AgitRepositoryUpdateRequest {
+                    title: Some(req.title),
+                };
+
+                match ctrl.repo.update(id, update_request).await {
+                    Ok(updated_agit) => Ok(Json(updated_agit)),
+                    Err(e) => Err(ServiceError::Unknown(format!("Failed to update agit: {}", e).into())),
+                }
```

Comment:
Firstly, I recommend it to be a separated function.

And you can simplize as below.
```suggestion
               match ctrl.repo.update(id, req.into()).await {
                    Ok(updated_agit) => Ok(Json(updated_agit)),
                    Err(e) => Err(ServiceError::Unknown(format!("Failed to update agit: {}", e).into())),
                }
```